src/esp/bindings/SensorBindings.cpp

```diff
@@ -248,7 +250,102 @@ void initSensorBindings(py::module& m) {
           R"(The distance to the near clipping plane for this CameraSensor uses.)")
       .def_property(
           "far_plane_dist", &CameraSensor::getFar, &CameraSensor::setFar,
-          R"(The distance to the far clipping plane for this CameraSensor uses.)");
+          R"(The distance to the far clipping plane for this CameraSensor uses.)")
+      .def(
+          "obs_buffer",
+          [](CameraSensor& self, int gpuDevice) {
+            py::handle handle = py::cast(self);
+            if (!py::hasattr(handle, "__obs_buffer")) {
+              if (self.specification()->gpu2gpuTransfer) {
+                auto torch = py::module_::import("torch");
+                if (self.specification()->sensorType ==
+                    esp::sensor::SensorType::Semantic) {
+                  py::setattr(
+                      handle, "__obs_buffer",
+                      torch.attr("empty")(
+                          (py::int_(self.specification()->resolution[0]),
+                           py::int_(self.specification()->resolution[1])),
+                          "dtype"_a = torch.attr("int32"),
+                          "device"_a = torch.attr("device")(
+                              py::str("cuda"), py::int_(gpuDevice))));
+                } else if (self.specification()->sensorType ==
+                           esp::sensor::SensorType::Depth) {
+                  py::setattr(
+                      handle, "__obs_buffer",
+                      torch.attr("empty")(
+                          (py::int_(self.specification()->resolution[0]),
+                           py::int_(self.specification()->resolution[1])),
+                          "dtype"_a = torch.attr("float32"),
+                          "device"_a = torch.attr("device")(
+                              py::str("cuda"), py::int_(gpuDevice))));
+                } else {
```

Comment:
This isn't future proof if we introduce a new SensorType, we should raise an error potentially if it's not an expected SensorType.

Reply:
I might actually just remove this for now. This was a carry over from the PR by that other former employee that I continued. I don't believe I am even using these bindings right now

Reply:
@Skylion007, like I said, I may just remove this section, but if I don't, would this be sufficient error catching?